Apps/Laboratory/src/Delegates/RestLaboratoryDelegate.cs

```diff
@@ -214,5 +223,98 @@ public async Task<RequestResult<LaboratoryReport>> GetLabReport(Guid id, string
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Maintainability", "CA1506:Avoid excessive class coupling", Justification = "Team decision")]
+        public async Task<RequestResult<PHSAResult<IEnumerable<PublicCovidTestResult>>>> GetPublicTestResults(string accessToken, string phn, DateOnly dateOfBirth, DateOnly collectionDate)
+        {
+            using Activity? activity = Source.StartActivity("GetPublicTestResults");
+            this.logger.LogDebug($"Getting public COVID-19 test results {phn} {dateOfBirth} {collectionDate}...");
+
+            HttpContent? content = null;
+            HttpContext? httpContext = this.httpContextAccessor.HttpContext;
+            string? ipAddress = httpContext?.Connection.RemoteIpAddress?.MapToIPv4().ToString();
+
+            RequestResult<PHSAResult<IEnumerable<PublicCovidTestResult>>> retVal = new()
+            {
+                ResultStatus = ResultType.Error,
+                PageIndex = 0,
+            };
+
+            try
+            {
+                JsonSerializerOptions serializerOptions = new()
+                {
+                    Converters = { new DateOnlyJsonConverter() },
+                };
+                string json = JsonSerializer.Serialize(new { phn, dateOfBirth, collectionDate }, serializerOptions);
+                using HttpClient client = this.httpClientService.CreateDefaultHttpClient();
+
+                string endpointString = $"{this.labConfig.BaseUrl}{this.labConfig.PublicCovidTestsEndPoint}";
+                Uri endpoint = new(endpointString);
+                content = new StringContent(json, null, MediaTypeNames.Application.Json);
+
+                client.DefaultRequestHeaders.Accept.Clear();
+                client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(MediaTypeNames.Application.Json));
+                client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("bearer", accessToken);
+                client.DefaultRequestHeaders.Add("X-Forward-For", ipAddress);
+
+                HttpResponseMessage response = await client.PostAsync(endpoint, content).ConfigureAwait(true);
+                string payload = await response.Content.ReadAsStringAsync().ConfigureAwait(true);
+                this.logger.LogTrace($"Response: {response}");
+
+                switch (response.StatusCode)
+                {
+                    case HttpStatusCode.OK:
+                        this.logger.LogTrace($"Response payload: {payload}");
+                        PHSAResult<IEnumerable<PublicCovidTestResult>>? phsaResult = JsonSerializer.Deserialize<PHSAResult<IEnumerable<PublicCovidTestResult>>>(payload);
+                        if (phsaResult != null && phsaResult.Result != null)
+                        {
+                            retVal.ResultStatus = ResultType.Success;
+                            retVal.ResourcePayload = phsaResult;
+                            retVal.TotalResultCount = 1;
```

Comment:
Total Result Count always 1?

Reply:
Good catch.  We only use this property for paginated results, but it doesn't hurt to populate it correctly.

Reply:
Fixed.